frontend/src/metabase/visualizations/echarts/cartesian/model/dataset.ts

```diff
@@ -270,3 +272,44 @@ export const getNormalizedDataset = (
     }, normalizedDatum);
   });
 };
+
+/**
+ * Calculates the minimum and maximum values (extents) for each series in the dataset.
+ *
+ * @param {DataKey[]} keys - The keys of the series to calculate extents for.
+ * @param {GroupedDataset} dataset - The dataset containing the series data.
+ * @returns {Record<DataKey, Extent>} Series extent by a series data key.
+ */
+export const getDatasetExtents = (
+  keys: DataKey[],
+  dataset: GroupedDataset,
+): Record<DataKey, Extent> => {
+  const extents: Record<DataKey, Extent> = {};
+
+  dataset.forEach(item => {
```

Comment:
```suggestion
  dataset.forEach(row => {
```

Nit: call it `row` instead of `item`